From 07a7a3a1c63609daa774051fa807f28d8b2a0450 Mon Sep 17 00:00:00 2001
From: Julien Marrec <julien.marrec@gmail.com>
Date: Tue, 19 May 2020 15:27:55 +0200
Subject: [PATCH] Repatch json.hpp like current

---
 third_party/nlohmann/json.hpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/third_party/nlohmann/json.hpp b/third_party/nlohmann/json.hpp
index 06da81532..7b43b9d80 100644
--- a/third_party/nlohmann/json.hpp
+++ b/third_party/nlohmann/json.hpp
@@ -47,6 +47,7 @@ SOFTWARE.
 #include <string> // string, stoi, to_string
 #include <utility> // declval, forward, move, pair, swap
 #include <vector> // vector
+#include <doj/alphanum.hpp>
 
 // #include <nlohmann/adl_serializer.hpp>
 
@@ -14931,9 +14932,9 @@ class basic_json
 #if defined(JSON_HAS_CPP_14)
     // Use transparent comparator if possible, combined with perfect forwarding
     // on find() and count() calls prevents unnecessary string construction.
-    using object_comparator_t = std::less<>;
+    using object_comparator_t = doj::alphanum_less<>;
 #else
-    using object_comparator_t = std::less<StringType>;
+    using object_comparator_t = doj::alphanum_less<StringType>;
 #endif
 
     /*!
-- 
2.26.2

